invenio_stats/utils.py

```diff
@@ -46,11 +52,14 @@ def get_bucket_size(client, index, agg_field):
             }
         }
     }
-    search = Search(using=client, index=index)
+    search = Search(using=client, index=index) \
+        .filter('range', timestamp=time_range)
```

Comment:
I think you need an `if time_range:` statement here, in case it is actually not used at all, i.e. when no `start_date`/`end_date` is passed.